lib/matplotlib/tests/test_colorbar.py

```diff
@@ -322,11 +321,8 @@ def test_colorbarbase():
 
 
 def test_parentless_mappable():
-    pc = mpl.collections.PatchCollection([], cmap=plt.get_cmap('viridis'))
-    pc.set_array([])
-
-    with pytest.warns(_api.MatplotlibDeprecationWarning,
-                      match='Unable to determine Axes to steal'):
+    pc = mpl.collections.PatchCollection([], cmap=plt.get_cmap('viridis'), array=[])
+    with pytest.raises(ValueError):
```

Comment:
Should keep the `match` on here.
```suggestion
    with pytest.raises(ValueError, match='Unable to determine Axes to steal'):
```